src/class-model.ts

```diff
@@ -251,23 +278,57 @@ export function register<Instance, Klass extends { new (...args: any[]): Instanc
   });
 
   // create the MST type for not-readonly versions of this using the views and actions extracted from the class
-  klass.mstType = mstTypes
+  let mstType = mstTypes
     .model(klass.name, mstPropsFromQuickProps(klass.properties))
     .views((self) => bindToSelf(self, mstViews))
     .actions((self) => bindToSelf(self, mstActions));
 
   if (Object.keys(mstVolatiles).length > 0) {
     // define the volatile properties in one shot by running any passed initializers
-    (klass as any).mstType = (klass as any).mstType.volatile((self: any) => initializeVolatiles({}, self, mstVolatiles));
+    mstType = mstType.volatile((self: any) => initializeVolatiles({}, self, mstVolatiles));
+  }
+
+  klass.snapshottedViews = metadatas.filter((metadata) => metadata.type == "snapshotted-view") as SnapshottedViewMetadata[];
+  if (klass.snapshottedViews.length > 0) {
+    // add a property to the MST type to track changes to a @snapshottedView when none of its model's properties changed
+    mstType = mstType
+      .props({ __snapshottedViewsEpoch: mstTypes.optional(mstTypes.number, 0) })
+      .actions((self) => ({ __incrementSnapshottedViewsEpoch: () => self.__snapshottedViewsEpoch++ }))
+      .actions((self) => {
+        const hook = isRoot(self) ? "afterCreate" : "afterAttach";
+        return {
+          [hook]() {
+            reaction(
+              () => {
+                return klass.snapshottedViews.map((sv) => {
+                  const value = self[sv.property];
+                  if (sv.options.createSnapshot) {
+                    return sv.options.createSnapshot(value);
+                  }
+                  if (Array.isArray(value)) {
+                    return value.map(getSnapshot);
+                  }
+                  return getSnapshot(value);
+                });
```

Comment:
Since we don't care about the data value, could we instead do all of this like this:

```js
for (const view of snapshottedViews) {
  reaction(
    () => {
      // what's inside of the `map`
    },
    () => {
      self.__incrementSnapshottedViewsEpoch();
    },
    { equals: comparer.structural },
  )
}
```

I'm thinking the benefit is that we much tighter dependency chains, avoiding the unnecessary creation of an array (from the `.map`), and trigger smaller payloads that `comparer.structural` has to look at.